capellambse_context_diagrams/collectors/makers.py

```diff
@@ -68,21 +81,37 @@ def make_diagram(diagram: context.ContextDiagram) -> _elkjs.ELKInputData:
 
 def make_label(
     text: str,
-    icon: tuple[int | float, int | float] = (0, 0),
+    icon: tuple[int | float, int | float] = (ICON_WIDTH, ICON_HEIGHT),
     layout_options: _elkjs.LayoutOptions | None = None,
-) -> _elkjs.ELKInputLabel:
+    max_width: int | float | None = None,
+) -> list[_elkjs.ELKInputLabel]:
     """Return an
     [`ELKInputLabel`][capellambse_context_diagrams._elkjs.ELKInputLabel].
     """
-    label_width, label_height = helpers.get_text_extent(text)
-    icon_width, icon_height = icon
-    layout_options = layout_options or DEFAULT_LABEL_LAYOUT_OPTIONS
-    return {
-        "text": text,
-        "width": icon_width + label_width + 2 * LABEL_HPAD,
-        "height": icon_height + label_height + 2 * LABEL_VPAD,
-        "layoutOptions": layout_options,
-    }
+    label_width, label_height = chelpers.get_text_extent(text)
+    icon_width, _ = icon
+    lines = [text]
+    if max_width is not None and label_width > max_width:
```

Comment:
We could add an Error message in case max_width is None. i.e. try max_width and continue with the logic if it applies.